login/service.go

```diff
@@ -636,44 +632,42 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateIdentityInDB(ctx context.Co
 	if err != nil {
 		log.Error(ctx, map[string]interface{}{
 			"err": err,
-		}, "unable to  query for an identity by username")
-		return nil, false, errors.New("Error during querying for an identity by ID " + err.Error())
+		}, "unable to query for an identity by username")
+		return nil, false, errs.Wrapf(err, "error during querying for an identity by ID")
 	}
 
 	if len(identities) == 0 {
 		return nil, false, autherrors.NewUnauthorizedError(fmt.Sprintf("user '%s' is not approved", userProfile.Username))
-	} else {
-		identity = &identities[0]
-
-		// we had done a
-		// keycloak.Identities.Query(account.IdentityFilterByID(keycloakIdentityID), account.IdentityWithUser())
-		// so, identity.user should have been populated.
+	}
+	identity = &identities[0]
 
-		if identity.User.ID == uuid.Nil {
-			log.Error(ctx, map[string]interface{}{
-				"identity_id": identity.ID,
-			}, "Found Keycloak identity is not linked to any User")
-			return nil, false, errors.New("found Keycloak identity is not linked to any User")
-		}
+	// we had done a
+	// keycloak.Identities.Query(account.IdentityFilterByID(keycloakIdentityID), account.IdentityWithUser())
+	// so, identity.user should have been populated.
 
-		if !identity.RegistrationCompleted {
-			newIdentityCreated = true
-			fillUserFromUserInfo(*userProfile, identity)
-			identity.RegistrationCompleted = true
-			err = transaction.Transactional(keycloak.App, func(tr transaction.TransactionalResources) error {
-				// Using the old-fashioned service
-				err := tr.Identities().Save(ctx, identity)
-				if err != nil {
-					return err
-				}
-				err = tr.Users().Save(ctx, &identity.User)
-				if err != nil {
-					return err
-				}
-				return nil
-			})
-		}
+	if identity.User.ID == uuid.Nil {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": identity.ID,
+		}, "Found Keycloak identity is not linked to any User")
```

Comment:
Technically, should not call it keycloak identity anymore :)

Reply:
oh right, let me fix that ;)